projects/backend/fronts.ts

```diff
@@ -3,52 +3,71 @@ import fromEntries from 'object.fromentries'
 
 export const getCollection = async (
     id: string,
     live: boolean = true,
     lastModifiedUpdater: LastModifiedUpdater,
-): Promise<Collection> => {
+): Promise<Attempt<Collection>> => {
     const resp = await attempt(
         s3fetch(`frontsapi/collection/${id}/collection.json`),
     )
-    if (hasFailed(resp)) throw new Error('Could not fetch from S3.')
+    if (hasFailed(resp))
+        return withFailureMessage(resp, 'Could not fetch from S3')
 
     lastModifiedUpdater(resp.lastModified)
 
     const deserialized = await attempt(resp.json())
-    if (hasFailed(resp)) throw new Error('Response was not valid json')
-
-    const collection: CollectionFromResponse = deserialized as CollectionFromResponse
-
-    const articleFragmentList = live ? collection.live : collection.draft
-    if (articleFragmentList == null) throw new Error('Hello that didnt work')
-    const articleFragments = fromEntries(
-        articleFragmentList.map(
-            (fragment): [string, NestedArticleFragment] => [
-                fragment.id,
-                fragment,
-            ],
-        ),
-    )
+    if (hasFailed(deserialized))
+        return withFailureMessage(deserialized, 'Response was not valid json')
+
+    const collection = deserialized as CollectionFromResponse
 
+    const articleFragmentList = collection.live.map(
+        (fragment): [number, NestedArticleFragment] => [
+            parseInt(fragment.id.replace('internal-code/page/', '')),
+            fragment,
+        ],
+    )
+    const articleFragments = fromEntries(articleFragmentList)
+    const ids: number[] = articleFragmentList.map(([id]) => id)
     const preview = live ? undefined : true
-    const capiArticles = await getArticles(Object.keys(articleFragments))
-    const articles: [string, Article][] = Object.entries(capiArticles).map(
-        ([key, article]) => {
-            const fragment =
-                articleFragments[`internal-code/page/${article.id}`] ||
-                articleFragments[key]
+    const capiPrintArticles = await attempt(getArticles(ids, 'printsent'))
+    const capiSearchArticles = await attempt(getArticles(ids, 'search'))
+    if (hasFailed(capiPrintArticles))
+        return withFailureMessage(
+            capiPrintArticles,
+            'Could not connect to capi print sent',
+        )
+    if (hasFailed(capiSearchArticles))
+        return withFailureMessage(
+            capiSearchArticles,
+            'Could not connect to CAPI',
+        )
+
+    const articles: [string, Article][] = Object.entries(articleFragments)
+        .filter(([key]) => {
+            const inResponse =
+                key in capiPrintArticles || key in capiSearchArticles
+            if (!inResponse) {
+                console.warn(`Removing ${key} as not in CAPI response.`)
+            }
+            return inResponse
+        })
+        .map(([key, fragment]) => {
+            const article = capiPrintArticles[key] || capiSearchArticles[key]
```

Comment:
This is great, but it probably makes sense the other way around. Imagine this... An article that is available via print-sent **and** real capi that is being actively edited; in this scenario we'd grab the dirty version, rather than the pristine.